onecodex/scripts/filter_reads.py

```diff
@@ -53,20 +53,61 @@ def write_fastx_record(record, handler):
     handler.write(record_str.format(*record))
 
 
-@click.command(help='Filter a FASTX file based on the taxonomic results from a CLASSIFICATION_ID')
-@click.argument('classification_id')
-@click.argument('fastx', type=click.Path())
-@click.option('-t', '--tax-id', required=True, multiple=True,
-              help='Filter to reads mapping to tax IDs. May be passed multiple times.')
-@click.option('-r', '--reverse', type=click.Path(), help='The reverse (R2) '
-              'read file, optionally')
-@click.option('--split-pairs/--keep-pairs', default=False, help='Keep only '
-              'the read pair member that matches the list of tax ID\'s')
-@click.option('-o', '--out', default='.', type=click.Path(), help='Where '
-              'to put the filtered outputs')
-@click.pass_context
-@pretty_errors
-def cli(ctx, classification_id, fastx, reverse, tax_id, split_pairs, out):
+def make_taxonomy_dict(classification, parent=False):
+    """
+    Takes a classification data frame returned by the API and parses it
+    into a dictionary mapping a tax_id to its children (or parent).
+    Restricted to tax_id's that are represented in the classification
+    results.
+    """
+
+    tax_id_map = {}
+
+    if parent:
+        for row in classification.table().itertuples():
```

Comment:
Subtle: It would be better to use `classification.results()` and iterate through those as the `.table()` method imports pandas which we do _not_ install by default. 

100% of CLI commands should work with a vanilla `pip install onecodex` vs. a full library install `pip install onecodex[all]`.

As needed we should also test this command in a tox environment _without_ all dependencies so we can track these. That may be worth setting up as part of this PR.

Reply:
Fixed.